### PR TITLE
test: add more tests for statistics reading

### PR DESCRIPTION
**Pull request by NGA-TRAN**
## Which issue does this PR close?

More tests for https://github.com/apache/datafusion/issues/10453

## Rationale for this change
Add tests for parquet statistics reading

## What changes are included in this PR?
 
Tests and bug tickets linked to appropriate tests

## Are these changes tested?

Yes

## Are there any user-facing changes?
No

### CODE REVIEW

**Comment by NGA-TRAN** [comment]
@alamb : This PR is ready for review

**Review by comphead — COMMENTED** [review]
thanks @NGA-TRAN good PR
Nice test description, this is important

What I still cannot understand is this a regression test for the bug we missed earlier?

**Comment by NGA-TRAN** [comment]
@comphead 

> What I still cannot understand is this a regression test for the bug we missed earlier?

I am working on new arrow statistics API https://github.com/apache/datafusion/issues/10453 and @alamb suggested me to add more coverage tests as well as moving tests for private functions in datafusion/core/src/datasource/physical_plan/parquet/statistics.rs to this file.  Sorry that I am not aware of available/related bug tickets

**Comment by alamb** [comment]
> What I still cannot understand is this a regression test for the bug we missed earlier?

My strong suspicion is that the bugs @NGA-TRAN  is finding would manifest themselves as potentially incorrect results when reading parquet files predicates on  these types of columns. 

It may also simply manifest as not being able to prune row groups based on such predicates

I haven't worked to make a reproducer as it would likely require creating parquet files with multiple row groups with carefully chosen data patterns

File: datafusion/core/tests/parquet/arrow_statistics.rs [hunk 1]
```diff
@@ -624,20 +624,281 @@ async fn test_dates_64_diff_rg_sizes() {
...
+    }
+    .run("f");
+}
+
+#[tokio::test]
+async fn test_decimal() {
+    let row_per_group = 5;
+    // This creates a parquet file of 1 column "decimal_col" with decimal data type and precicion 9, scale 2
+    // file has 3 record batches, each has 5 rows. They will be saved into 3 row groups
+    let reader = parquet_file_many_columns(Scenario::Decimal, row_per_group).await;
```
Comment (alamb) [inline datafusion/core/tests/parquet/arrow_statistics.rs:748]: Another important thing to test with decimals is different precision / scales -- maybe we can do this as a different PR

File: datafusion/core/tests/parquet/arrow_statistics.rs [hunk 2]
```diff
@@ -624,20 +624,281 @@ async fn test_dates_64_diff_rg_sizes() {
...
+
+    // column "service_binary"
+    let reader = parquet_file_many_columns(Scenario::ByteArray, row_per_group).await;
+    Test {
+        reader,
+        expected_min: Arc::new(StringArray::from(vec![
+            "frontend five",
+            "backend one",
+            "backend eight",
+        ])), // Shuld be BinaryArray
```
Comment (alamb) [inline datafusion/core/tests/parquet/arrow_statistics.rs:828]: 👍

**Review by alamb — APPROVED** [review]
Thank you @NGA-TRAN  -- I think this is excellent test coverage

**Comment by alamb** [comment]
Since this PR is just tests, merging it in

File: datafusion/core/tests/parquet/arrow_statistics.rs (hunk 1 above)
Comment (NGA-TRAN) [reply datafusion/core/tests/parquet/arrow_statistics.rs:748]: @alamb : can you be more specific? In the test below, I had to make sure they have the same precision & scale. What else do we have to test here?
